Source/Core/SwipeActions.swift

```diff
@@ -0,0 +1,134 @@
+//
+//  Swipe.swift
+//  Eureka
+//
+//  Created by Marco Betschart on 14.06.17.
+//  Copyright © 2017 Xmartlabs. All rights reserved.
+//
+
+import Foundation
+
+public typealias SwipeActionHandler = (Any, Any, ((Bool) -> Void)?) -> Void
+
+
+public class SwipeAction{
+	
+	public let platformValue: Any
+	
+	public var backgroundColor: UIColor?{
+		didSet{
+			if #available(iOS 11, *), let platformValue = self.platformValue as? UIContextualAction{
+				platformValue.backgroundColor = self.backgroundColor
+			}
+		}
+	}
+	
+	public var image: UIImage?{
+		didSet{
+			if #available(iOS 11, *), let platformValue = self.platformValue as? UIContextualAction{
+				platformValue.image = self.image
+			}
+		}
+	}
+	
+	public var title: String?{
+		didSet{
+			if #available(iOS 11, *), let platformValue = self.platformValue as? UIContextualAction{
+				platformValue.title = self.title
+			}
+		}
+	}
+	
+	public let handler: SwipeActionHandler
+	public let style: Style
+	
+	public init(style: Style, title: String?, handler: @escaping SwipeActionHandler){
+		self.style = style
+		self.handler = handler
+		self.title = title
+		
+		if #available(iOS 11, *){
+			self.platformValue = UIContextualAction(style: style.platformValue as! UIContextualAction.Style, title: title){ action, view, completion -> Void in
+				handler(action,view,completion)
+			}
+			
+		} else {
+			self.platformValue = UITableViewRowAction(style: style.platformValue as! UITableViewRowActionStyle,title: title){ (action, indexPath) -> Void in
+				handler(action,indexPath,nil)
+			}
+		}
+	}
+	
+	public enum Style: Int{
+		case normal = 0
+		case destructive = 1
+		
+		var platformValue: Any{
```

Comment:
Would be great if all these `platformValues` can have a type more concrete than `Any`. @mtnbarreto thoughts?

Reply:
Good point. Used marker protocols to fix this. See `SwipeActions.swift` in the bottom of the file.